### PR TITLE
Update thanos engine and Prometheus dependencies

### PR DESCRIPTION
**Pull request by fpetkovski**
This commit bumps thanos/promql-engine to latest main and resolves breaking changes from the prometheus/prometheus dependency.

* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

Update: 
* thanos/promql-engine to latest main
* prometheus/prometheus to 2.46.1

## Verification

Relying on tests.

### CODE REVIEW

**Comment by MichaHoffmann** [comment]
Oh, can we merge thsi with https://github.com/thanos-io/thanos/pull/6648 together?  I have some wal compression fixes too there

**Review by MichaHoffmann — APPROVED** [review]
LGTM

**Comment by fpetkovski** [comment]
Ah sorry, completely missed that PR. Maybe we can merge this one since the tests are now passing, and we can use the other one to enable subqueries in the engine.

**Comment by MichaHoffmann** [comment]
> Ah sorry, completely missed that PR. Maybe we can merge this one since the tests are now passing, and we can use the other one to enable subqueries in the engine.

Yeah lets do that ill just do the small changes in querier in a followup PR!

File: pkg/store/tsdb_test.go [hunk 1]
```diff
@@ -585,7 +586,7 @@ func benchTSDBStoreSeries(t testutil.TB, totalSamples, totalSeries int) {
 
 	for j := 0; j < 3; j++ {
 		head, created := storetestutil.CreateHeadWithSeries(t, j, storetestutil.HeadGenOptions{
-			TSDBDir:          tmpDir,
+			TSDBDir:          t.TempDir(),
```
Comment (GiedriusS) [inline pkg/store/tsdb_test.go:589]: `tmpDir` is already `t.TempDir()`, why this change?

Comment (MichaHoffmann) [reply pkg/store/tsdb_test.go:589]: Oh, subsequent calls to "t.Tempdir" return different directories; is that a bug?

Comment (fpetkovski) [reply pkg/store/tsdb_test.go]: I using the same local dir caused the tests to fail after upgrading to 2.46, but it does not seem to be the case. It was rather because we close the head and entire db in a non-deterministic manner by using `t.CleanUp` for one, and `defer` for the other.

**Review by GiedriusS — APPROVED** [review]
